packages/peregrine/lib/talons/MagentoRoute/useMagentoRoute.js

```diff
@@ -0,0 +1,63 @@
+import { useEffect, useRef, useState } from 'react';
+import { useLocation } from 'react-router-dom';
+import { useApolloClient } from '@apollo/react-hooks';
+
+import addToCache from './addToCache';
+import getRouteComponent from './getRouteComponent';
+
+const CLIENT_PATHS = new Set().add('/search.html');
```

Comment:
One last thing, it feels weird that the list of client-only routes is kept here and in `Routes`.

```jsx
<Suspense fallback={fullPageLoadingIndicator}>
    <Switch>
        <Route exact path="/search.html">
            <Search />
        </Route>
        <Route exact path="/create-account">
            <CreateAccountPage />
        </Route>
        <Route>
            <MagentoRoute />
        </Route>
    </Switch>
</Suspense>
```

Don't we know by the fact we made it down to `MagentoRoute` that we're not a client-only route?

Reply:
Also, if `/search.html` is here, should `/create-account` be too?

Reply:
I agree. Of course, I'm not the one who carved out the exception for `search.html` here, so I don't know why that clause is here when it appears to be unreachable. Worth investigating.

Reply:
@supernova-at Looks like you were the one who added the clause for `/search.html` to `MagentoRouteHandler` in #639, almost a year ago! In fact, [here's what I had to say about it](https://github.com/magento/pwa-studio/pull/639/files#r241900060) at the time: 😅 

> This pattern doesn't seem like it will scale well. Are we going to add a new condition for every client-side route?

I also found another [fun comment I left](https://github.com/magento/pwa-studio/pull/639#issuecomment-449088676) on that PR.

> Now that React Router uses the new context API, our context work in Router is probably redundant. The only additional property being passed by our context is `apiBase`, which could probably be an environment variable or something.

Fixed!

Reply:
Also, fixed in [cabd8ec](https://github.com/magento/pwa-studio/pull/1966/commits/cabd8ecc94dab18501fde1ea7d722e6355b54c1c).